Foreign/LibFFI/Base.hs

```diff
@@ -49,6 +49,23 @@ newStructCType cTypes = do
     init_ffi_type ffi_type elements
     return (ffi_type, free ffi_type >> free elements)
 
+sizeAndAlignmentOfCType :: Ptr CType -> IO (Int, Int)
+sizeAndAlignmentOfCType cType = do
+  (size, alignment) <- ffi_type_size_and_alignment cType
+  if size /= 0 && alignment /= 0
+  then return (fromIntegral size, fromIntegral alignment)
+  else do
+    -- The type's size and alignment haven't been initialized
+    -- so we force it with a call to `ffi_get_struct_offsets`.
```

Comment:
```suggestion
    -- so we force it with a call to `ffi_prep_cif`.
```

Reply:
Oops yeah, I ended writing the second commit twice because I messed up my local state. So I forgot to fix the comment the second time around. Thanks for pointing it out.